server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java

```diff
@@ -0,0 +1,116 @@
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.apache.gravitino.GravitinoEnv;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.MetadataObjects;
+import org.apache.gravitino.authorization.Owner;
+import org.apache.gravitino.authorization.OwnerManager;
+import org.apache.gravitino.dto.requests.OwnerSetRequest;
+import org.apache.gravitino.dto.responses.OwnerResponse;
+import org.apache.gravitino.dto.responses.SetResponse;
+import org.apache.gravitino.dto.util.DTOConverters;
+import org.apache.gravitino.metrics.MetricNames;
+import org.apache.gravitino.server.authorization.NameBindings;
+import org.apache.gravitino.server.web.Utils;
+
+@NameBindings.AccessControlInterfaces
+@Path("/metalakes/{metalake}/owners")
+public class OwnerOperations {
+
+  private final OwnerManager ownerManager;
+
+  @Context private HttpServletRequest httpRequest;
+
+  public OwnerOperations() {
+    // Because ownerManager may be null when Gravitino doesn't enable authorization,
+    // and Jersey injection doesn't support null value. So OwnerOperations chooses to retrieve
+    // ownerManager from GravitinoEnv instead of injection here.
+    this.ownerManager = GravitinoEnv.getInstance().ownerManager();
+  }
+
+  @GET
+  @Path("{type}/{fullName}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "get-object-owner." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "get-object-owner", absolute = true)
+  public Response getOwnerForObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName) {
+    try {
+      MetadataObject object =
+          MetadataObjects.parse(
+              fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            Optional<Owner> owner = ownerManager.getOwner(metalake, object);
+            if (owner.isPresent()) {
+              return Utils.ok(new OwnerResponse(DTOConverters.toDTO(owner.get())));
+            } else {
+              return Utils.ok(new OwnerResponse(null));
+            }
+          });
+    } catch (Exception e) {
+      return ExceptionHandlers.handleOwnerException(
+          OperationType.GET, String.format("metadata object %s", fullName), metalake, e);
+    }
+  }
+
+  @PUT
+  @Path("{type}/{fullName}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "set-object-owner." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "set-object-owner", absolute = true)
+  public Response setOwnerForObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName,
+      OwnerSetRequest request) {
```

Comment:
What's the difference between `type` in  `OwnerSetRequest`  and type in line 97.

Reply:
One is the type of owner, the other is the type of metadata object.

Reply:
Can you rename the parameter to `owerType` or something similar?

Reply:
I hesitate to add this. I worry whether it is clear enough. The type of `type` is Owner.Type. Is it enough?

Reply:
Change 97Line type to metadataObjectType.